### PR TITLE
Read receipts

### PR DESCRIPTION
**Pull request by dbkr**
Bit of a mess of a PR separating out user activity from presence so we can also use it for read receipts. Also contains some bits of my member / room avatar code because apparently I committed them on the wrong branch...

### CODE REVIEW

File: src/UserActivity.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ */
+class UserActivity {
+
+    /**
+     * Start listening to user activity
+     */
+    start() {
+        document.onmousemove = this._onUserActivity.bind(this);
+        document.onkeypress = this._onUserActivity.bind(this);
+        this.lastActivityAt = (new Date).getTime();
```
Comment (Kegsay) [inline src/UserActivity.js]: I've not seen this syntax before... any reason why not `new Date().getTime()` which is more Javay? I would also love if the variable mentioned `timestamp` or `ts` or something in it to make it clearer it's a timestamp.

File: src/controllers/organisms/RoomView.js [hunk 2]
```diff
@@ -346,7 +359,7 @@ module.exports = {
                 }
             }
             ret.unshift(
-                <li key={mxEv.getId()}><EventTile mxEvent={mxEv} continuation={continuation} last={last}/></li>
+                <li ref={this._collectEventNode.bind(this, mxEv.getId())} key={mxEv.getId()}><EventTile mxEvent={mxEv} continuation={continuation} last={last}/></li>
```
Comment (Kegsay) [inline src/controllers/organisms/RoomView.js]: I thought you could only assign `ref` to a string? What does this do?

**Comment by Kegsay** [comment]
From what I can deduce it LGTM, but it's hard to tell how it all fits together from this PR alone.

File: src/UserActivity.js (hunk 1 above)
Comment (dbkr) [inline src/UserActivity.js]: Honestly I think I probably just got it from googling as unsure what the correct precedence is in JS for new / dot operator. Your suggestion is probably clearer to me.

File: src/controllers/organisms/RoomView.js (hunk 2 above)
Comment (dbkr) [inline src/controllers/organisms/RoomView.js]: refs as strings is apparently kind of deprecated in react now, and they should be functions that get called with the react node once it gets mounted.

**Pull request merged**